e2e-tests/e2eTest.spec.ts

```diff
@@ -25,88 +25,193 @@ const iccidOld = getRandomICCID(4573)
 const iccidNewWL = getRandomICCID(4446)
 const iccidOldWL = getRandomICCID(4446)
 const iccidNotExisting = getRandomICCID(4573)
+const iccidNoHistory = getRandomICCID(4446)
 const now = new Date()
 const sixMinAgo = new Date(Date.now() - 6 * 60 * 1000)
 const timestampsLastHour = getTimestampsForSeeding(60, 5)
 const timestampsLastDay = getTimestampsForSeeding(60 * 24, 60)
-const randomUsedBytesVal = [1, 3, 67, 1, 2, 3, 5, 7, 2, 1, 42, 4]
-const randomUsedBytesVal2 = [5, 3, 1, 7, 89, 3, 4, 1, 3, 7, 0, 0]
+const timestampsLastTwoDays = getTimestampsForSeeding(60 * 24 * 1.5, 60 * 4)
+const usedBytes = [1, 3, 67, 1, 2, 3, 5, 7, 2, 1, 42, 4]
+const usedBytes2 = [5, 3, 1, 7, 89, 3, 4, 1, 3, 7, 0, 0]
+const usedBytesLastTwoDays = [1, 2, 3, 4, 5, 6, 7, 8, 9]
+
+const usageLastHour = timestampsLastHour.map((ts, index) => ({
+	ts,
+	usedBytes: usedBytes[index] ?? 0,
+}))
+
+const usageLastHour2 = timestampsLastHour.map((ts, index) => ({
+	ts,
+	usedBytes: usedBytes2[index] ?? 0,
+}))
+
+const usageLastDay = timestampsLastDay.map((ts, index) => ({
+	ts,
+	usedBytes: usedBytes[index] ?? 0,
+}))
+
+const usageLastTwoDays = timestampsLastTwoDays.map((ts, index) => ({
+	ts,
+	usedBytes: usedBytesLastTwoDays[index] ?? 0,
+}))
 
 const seedTimestreamFunc = seedTimestream(tsw)
 const fetchDataFunc = fetchData(APIURL)
-void describe('e2e-tests', () => {
+void describe('e2e-tests', async () => {
 	before(async () => {
 		//put notExisting SIM in DB
 		await putSimDetails(
 			db,
 			outputs.cacheTableName,
 		)({ iccid: iccidNotExisting, simExisting: false })
-		//put iccid from vendors in DB
+		//put noHistory SIM in DB
 		await seedingDBFunction({
-			iccidNew,
-			iccidOld,
-			now,
-			sixMinAgo,
+			iccid: iccidNoHistory,
+			usageTimestamp: now,
+			simDetails: { usedBytes: 0, totalBytes: 1000 },
 		})
+	})
+	void it(`should return statusCode 200, cache max-age=300 and correct body for iccid: ${iccidNewWL}`, async () => {
 		await seedingDBFunction({
-			iccidNew: iccidNewWL,
-			iccidOld: iccidOldWL,
-			now,
-			sixMinAgo,
+			iccid: iccidNewWL,
+			usageTimestamp: now,
+			simDetails: { usedBytes: 0, totalBytes: 1000 },
 		})
-		await seedTimestreamFunc(
-			timestampsLastHour,
-			randomUsedBytesVal,
-			iccidNewWL,
+		await seedTimestreamFunc({
+			usage: usageLastHour,
```

Comment:
*problem:* Please don't use magic here, seed the database with the actual values, make them inline:

```typescript
await seedDbWith([
{value: 67, ts: Date}
])
```

so it becomes clear which data is present in the DB, without having to jump back and forth.

Reply:
I understand that this would make it easier to read, but I don't understand how I can make the actual values (ts) inline? I need to use the ts for 'lastHour', 'lastDay' and 'lastTwoDays'. This can not be hardcoded; and I need to check it later in the e2e test meaning I need to make it a const outside of the test case? Or am I missing something?

Reply:
You still can initialize these values in the test case:

```typescript

// Setup
const lastHour = getLastHourDate()
await seedDb([
{value: 67, ts: lastHour}
])

// Execute
const result = await query(...)

// Assert
...
```